SOSS/extract/engine_legacy.py

```diff
@@ -0,0 +1,311 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+# General imports.
+import numpy as np
+from scipy.interpolate import interp1d, RectBivariateSpline
+
+# Astronoomy imports.
+from astropy.io import fits
+
+# Plotting.
+import matplotlib.pyplot as plt
+from matplotlib.colors import LogNorm
+
+###############################################
+# Hack to get the path of module. To be changed.
+from os.path import abspath, dirname
+
+
+def get_module_path(file):
```

Comment:
maybe a #TODO to remove this?

Reply:
The entire engine_legacy.py file should become obsolete in the near future. It is only there because I did not want to break some existing examples.